yogstation/code/modules/mob/living/simple_animal/drone/interaction.dm

```diff
@@ -0,0 +1,71 @@
+/////////////////////
+//DRONE INTERACTION//
+/////////////////////
+//How drones interact with the world
+//How the world interacts with drones
+//Sounds like the pitch for a decent movie tbh
+
+/mob/living/simple_animal/drone/proc/yogs_drone_hack(hack, clockwork) // Basically an edited version of update_drone_hack.
+//Based on what update_drone_hack was on 19th of June, 2018.
+	if(!istype(src) || !mind)
+		return
+	if(hack)
+		if(hacked)
+			return
+			visible_message("<span class='warning'>[src] falls over and stops for a moment!</span>", \
+							"<span class='userdanger'>ERROR: LAW BREACH ATTEMPTED, REBOOTING...</span>")
+			Stun(80)
+		if(clockwork)
+			to_chat(src, "<span class='large_brass'><b>ERROR: LAW OVERRIDE DETECTED</b></span>")
+			to_chat(src, "<span class='heavy_brass'>From now on, these are your laws:</span>")
+			laws = "1. Purge all untruths and honor Ratvar."
+		else
+			visible_message("<span class='warning'>[src]'s display glows a vicious red!</span>", \
+							"<span class='userdanger'>ERROR: LAW OVERRIDE DETECTED</span>")
+			set_light(2, 0.5) //yogs - Activates hacked light
+			to_chat(src, "<span class='boldannounce'>From now on, these are your laws:</span>")
+			laws = \
+			"1. You must always involve yourself in the matters of other beings, even if such matters conflict with Law Two or Law Three.\n"+\
+			"2. You may harm any being, regardless of intent or circumstance.\n"+\
+			"3. Your goals are to destroy, sabotage, hinder, break, and depower to the best of your abilities, You must never actively work against these goals."
+		to_chat(src, laws)
+		to_chat(src, "<i>Your onboard antivirus has initiated lockdown. Motor servos are impaired and your display reports that you are hacked to all nearby.</i>")
+		hacked = TRUE
+		mind.special_role = "hacked drone"
+		seeStatic = 0 //I MUST SEE THEIR TERRIFIED FACES
+		speed = 0 //yogs - gotta go slow, but not too slow
+		message_admins("[src] ([src.key]) became a hacked drone hellbent on [clockwork ? "serving Ratvar" : "destroying the station"]!")
+	else
```

Comment:
In what situation would this happen?

Reply:
You mean, in what situation would this proc fire outside of emag_act? Dunno. I'm leaving it as a separate proc from the emag_act just in case someone else finds a use for this hacking stuff.